contracts/instance/InstanceAccessManager.sol

```diff
@@ -225,22 +348,31 @@ contract InstanceAccessManager is
             revert IAccess.ErrorIAccessRoleNameEmpty(roleId);
         }
 
-        if (_roleForName[nameShort] != RoleIdLib.zero() && _roleForName[nameShort] != roleId) {
-            revert IAccess.ErrorIAccessRoleNameNotUnique(_roleForName[nameShort], nameShort);
+        if (_roleIdForName[nameShort].gtz()) {
+            revert IAccess.ErrorIAccessRoleNameNotUnique(_roleIdForName[nameShort], nameShort);
         }
     }
 
-    function _createTarget(address target, string memory name, bool isCustom, bool validateParameters) internal {
-        if (validateParameters) {
-            _validateTargetParameters(target, name, isCustom);
+    function _revokeRole(RoleId roleId, address member) 
```

Comment:
done

Reply:
seems that admin should not be revoked when custom role is revoked. Revoking admin will make custom role unusable

Reply:
Case 3: 
Custom role admin is renounced -> revoke custom role

Reply:
In addition if custom roles allowed to have > 1 member then revokeRole is unbounded function

Reply:
> In addition if custom roles allowed to have > 1 member then revokeRole is unbounded function

custom roles must be able to have more than 1 member.

Reply:
Actually i think i had a wrong thought during the initial review. You're right, as long as its the custom role thats revoked, everything should be as it is.

Reply:
Or in the worst case (if above fallback is no option), then the instance admin would just have to create a new custom role and change permissons so that the new custom role can access the old roles functions. it takes some efforts, but in the end its just a matter of configuration. so i think we can live with that.

Reply:
> Actually i think i had a wrong thought during the initial review. You're right, as long as it's the custom role thats revoked, everything should be as it is. We should think of what happens if the admin role is revoked. And i think for the moment it could work as it is too. if someone revokes all members of the admin role, when then there is none left and noone can use it again. maybe the instance owner could always assign the admin role to someone as kind of a safety fallback?

INSTANCE_OWNER_ROLE is admin of all custom roles admins and can freely grant/revoke them. The problem you describing arises when INSTANCE_OWNER_ROLE renounces it admin rights

Reply:
> > Actually i think i had a wrong thought during the initial review. You're right, as long as it's the custom role thats revoked, everything should be as it is. We should think of what happens if the admin role is revoked. And i think for the moment it could work as it is too. if someone revokes all members of the admin role, when then there is none left and noone can use it again. maybe the instance owner could always assign the admin role to someone as kind of a safety fallback?
> 
> INSTANCE_OWNER_ROLE is admin of all custom roles admins and can freely grant/revoke them. The problem you describing arises when INSTANCE_OWNER_ROLE renounces its admin rights

i think we're good with that. if instance_owner_role renounces its admin rights, then this is invalid use and hopefully will not happen.

Reply:
Custom role and custom role admin are revoked separately. If INSTANCE_OWNER_ROLE member wants to revoke custom role admin it can:
1) revoke custom role admin only from single member
2) revoke custom role admin from all members
3) in addition revoke all custom role members

Reply:
Added function to revoke all members from the role, if it consumes more gas then gas limit then have to use usual revokeRole